backend/src/middleware/notifications/notificationsMiddleware.spec.js

```diff
@@ -541,5 +571,148 @@ describe('notifications', () => {
         })
       })
     })
+
+    describe('given I file a report on a', () => {
+      let resourceId
+      const reasonCategory = 'discrimination_etc'
+      const reasonDescription = 'I am free to be gay !!!'
+      let reportFiler
+      let reportedUserOrAuthorData
+      const fileReportAction = async () => {
+        authenticatedUser = await reportFiler.toJson()
+        await mutate({
+          mutation: fileReportMutation,
+          variables: {
+            resourceId,
+            reasonCategory,
+            reasonDescription,
+          },
+        })
+        authenticatedUser = await notifiedUser.toJson()
+      }
+      const setExpectedNotificationOfReportedResource = reportedResource => {
+        return expect.objectContaining({
+          data: {
+            notifications: [
+              {
+                createdAt: expect.any(String),
+                read: false,
+                reason: 'filed_report_on_resource',
+                from: {
+                  __typename: 'Report',
+                  id: expect.any(String),
+                  filed: [
+                    {
+                      reasonCategory: 'discrimination_etc',
+                      reasonDescription: 'I am free to be gay !!!',
+                      reportedResource,
+                    },
+                  ],
+                },
+              },
+            ],
+          },
+        })
+      }
+
+      beforeEach(async () => {
+        reportFiler = notifiedUser
+        reportedUserOrAuthorData = {
+          id: 'reportedUser',
+          name: 'Mrs Badman',
+          slug: 'mrs-badman',
+          email: 'reported-user@example.org',
+          password: '1234',
+        }
+      })
+
+      describe('user', () => {
+        it('sends me a notification for filing a report on a user', async () => {
+          await factory.create('User', reportedUserOrAuthorData)
+          resourceId = 'reportedUser'
+          await fileReportAction()
+
+          const expected = setExpectedNotificationOfReportedResource({
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-01-23T18:38:35Z" title="Thursday, January 23rd 2020, 7:38:35 pm +01:00">Jan 23, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -541,5 +573,148 @@ describe('notifications', () => {
         })
       })
     })
+
+    describe('given me filing a report of a', () => {
+      let resourceId
+      const reasonCategory = 'discrimination_etc'
+      const reasonDescription = 'I am free to be gay !!!'
+      let reportFiler
+      let reportedUserOrAuthorData
+      const fileReportAction = async () => {
+        authenticatedUser = await reportFiler.toJson()
+        await mutate({
+          mutation: fileReportMutation,
+          variables: {
+            resourceId,
+            reasonCategory,
+            reasonDescription,
+          },
+        })
+        authenticatedUser = await notifiedUser.toJson()
+      }
+      const setExpectedNotificationOfReportedResource = reportedResource => {
+        return expect.objectContaining({
+          data: {
+            notifications: [
+              {
+                createdAt: expect.any(String),
+                read: false,
+                reason: 'filed_report_on_resource',
+                from: {
+                  __typename: 'Report',
+                  id: expect.any(String),
+                  filed: [
+                    {
+                      reasonCategory: 'discrimination_etc',
+                      reasonDescription: 'I am free to be gay !!!',
+                      reportedResource,
+                    },
+                  ],
+                },
+              },
+            ],
+          },
+        })
+      }
+
+      beforeEach(async () => {
+        reportFiler = notifiedUser
+        reportedUserOrAuthorData = {
+          id: 'reportedUser',
+          name: 'Mrs Badman',
+          slug: 'mrs-badman',
+          email: 'reported-user@example.org',
+          password: '1234',
+        }
+      })
+
+      describe('user', () => {
+        it('sends me a notification for filing a report on a user', async () => {
+          await neode.create('User', reportedUserOrAuthorData)
+          resourceId = 'reportedUser'
+          await fileReportAction()
+
+          const expected = setExpectedNotificationOfReportedResource({
```


I find this quite interesting :blush: ... on the right we have a super verbose variable name whereas the left we have a very generic one... :smile_cat:

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2020-01-24T09:25:00Z" title="Friday, January 24th 2020, 10:25:00 am +01:00">Jan 24, 2020</time>_

---


Hehe …